src/net_processing.cpp

```diff
@@ -3110,7 +3156,7 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
             }
 
             if (inv.IsMsgBlk()) {
-                const bool fAlreadyHave = AlreadyHaveBlock(inv.hash);
+                const bool fAlreadyHave = AlreadyHaveBlock(pfrom, inv.hash);
                 LogPrint(BCLog::NET, "got inv: %s  %s peer=%d\n", inv.ToString(), fAlreadyHave ? "have" : "new", pfrom.GetId());
```

Comment:
This log can now be misleading if we receive a block INV for a block we already have, but the peer shouldn't know about, since it'll say "new".

Reply:
Changed it to "maybe new", is that better?

In this location we are not able to tell from the result of `LookupBlockIndexForPeer` whether the index did not exist or if the peer should not know about it. I could add logging `LookupBlockIndexForPeer`, what do you think about that?